services/web/server/src/simcore_service_webserver/catalog_handlers_reverse_proxy.py

```diff
@@ -36,10 +36,15 @@ async def reverse_proxy_handler(request: web.Request) -> web.Response:
     # FIXME: hack
     if "/services" in backend_url.path:
         backend_url = backend_url.update_query({"user_id": user_id})
-    logger.debug("Redirecting '%s' -> '%s'", request.url, backend_url)
+    logger.debug(
+        "Redirecting '%s' -> '%s'",
+        request.url,
+        backend_url,
+        extra=get_log_record_extra(user_id=user_id),
```

Comment:
so the idea is to pass the user_id in the log, which is good.
Now I am not sure the way this is done is very usable, I expect that people will forget that very easily.
ideally the logger would check in the context if there is a user_id variable. but that is probably not really possible... you can get the variables, and maybe check if there is one called "user_id". but I agree that might be a bit fishy.

Reply:
Yes, I agree :+1:  , in the beginning, I was looking into some options on how to utilize context vars for that, but in the first run it didn't work out, therefore didn't want to overspend time on it. I would say this is the first version of how we can start to use it and see whether it makes sense (whether it will be useful for somebody), and it that case we can expand with the context vars or also with additional fields such as node_id/project_id. We can always remove it if it didn't bring any value.